### DISCUSSION TITLE
Enable the plugin by default

### DISCUSSION THREAD

**Discussion by amaltv** [microsoft/semantic-kernel#9464] [Q&A] [answered]
Do we have a way to enable the plugins by default? Currently each time I am adding the plugin and enabling it.. Can we enabled for the users by default?

**Comment by moonbox3** [comment]
Hello, @amaltv, thanks for your question. Which SK Language are you using?

**Comment by amaltv** [comment]
Hi , C# for the kernel and custom plugins are developed with python flask
in open api spec

**Reply by moonbox3** [reply]
Thanks for your response. I am assuming when you say "adding the plugin and enabling it..." you're referring to having to configure the `FunctionChoiceBehavior` on the prompt execution settings as shown in [these concept samples](https://github.com/microsoft/semantic-kernel/blob/main/dotnet/samples/Concepts/FunctionCalling/FunctionCalling.cs)? 

As any request to an AI service already requires the prompt execution settings to follow, one would indeed need to configure the these settings with the `FunctionChoiceBehavior` that they desire so we know which plugins to advertise to the model. It's safer to do this 'opt-in' approach versus an 'opt-out' approach of always sending tools to the model without a developer needing to configure which tools they want sent.

**Reply by amaltv** [reply]
Thanks @moonbox3 for the details.
We are using this repo : https://github.com/microsoft/chat-copilot to create a chat- copilot. 
In this repo there is one configuration file : https://github.com/microsoft/chat-copilot/blob/main/webapi/appsettings.json and in this file we can add all the custom plugins under the  "Plugins": section.. We have done that its loading in the chatbot page.. but its not enabled.. We need to go an enable it manually by clicking on the Enable button. Do we have any option to auto enable this as we cannot expect our user to do this manually.

**Comment by alliscode** [comment, marked as answer]
Hi @amaltv, to enable the plugins by default, take a look at the README on adding custom plugins. https://github.com/microsoft/chat-copilot/tree/main/webapi#adding-custom-plugins

Thanks!

**Reply by amalthomasintel** [reply]
@alliscode I was referring to the third-party plugins which are hosted outside: https://github.com/microsoft/chat-copilot/blob/main/plugins/README.md

This is what mentioned in the configuration settings. When we do this plugins are added but not enabled by default.. Can we enable this by default?
Is there any attribute called enabled: true?

I am pretty new to this, it would be great if you can guide on this problem

// A list of plugins that will be loaded by the application.
  // - Name is the NameForHuman of the plugin.
  // - ManifestDomain is the root domain of the plugin: https://platform.openai.com/docs/plugins/production/domain-verification-and-security
  // - Key is the key used to access the plugin if it requires authentication.
  "Plugins": [
    // Klarna Shopping does not require authentication.
    {
      "Name": "Klarna Shopping",
      "ManifestDomain": "https://www.klarna.com"
      // "Key": ""
    }
  ],
